src/library/autodj/autodjprocessor.cpp

```diff
@@ -756,19 +773,23 @@ void AutoDJProcessor::playerPositionChanged(DeckAttributes* pAttributes,
     // - transition into fading mode, play the other deck and fade to it.
     // - check if fading is done and stop the deck
     // - update the crossfader
-    if (thisPlayPosition >= thisDeck->fadeBeginPos && thisDeck->isFromDeck) {
+    if (thisPlayPosition >= thisDeck->fadeBeginPos && thisDeck->isFromDeck && !otherDeck->loading) {
         if (m_eState == ADJ_IDLE) {
             if (thisDeckPlaying || thisPlayPosition >= 1.0) {
                 // Set the state as FADING.
                 m_eState = thisDeck->isLeft() ? ADJ_LEFT_FADING : ADJ_RIGHT_FADING;
                 m_transitionProgress = 0.0;
                 emitAutoDJStateChanged(m_eState);
 
+                const double toDeckFadeDisatance =
+                        (thisDeck->fadeEndPos - thisDeck->fadeBeginPos) *
+                        getEndSecond(thisDeck) / getEndSecond(thisDeck);
+                // Re-cue the track if the user has seeked forward and will miss the fadeBeginPos
+                if (otherDeck->playPosition() >= otherDeck->fadeBeginPos - toDeckFadeDisatance) {
+                    otherDeck->setPlayPosition(otherDeck->startPos);
+                }
```

Comment:
typo and optimization.
```suggestion
                const double toDeckFadeDistance =
                        (thisDeck->fadeEndPos - thisDeck->fadeBeginPos);
                // Re-cue the track if the user has seeked forward and will miss the fadeBeginPos
                if (otherDeck->playPosition() >= otherDeck->fadeBeginPos - toDeckFadeDistance) {
                    otherDeck->setPlayPosition(otherDeck->startPos);
                }
```
I also find the `thisDeck->fadeEndPos - thisDeck->fadeBeginPos` very brittle. We should introduce methods on the deck or track class that returns the intro/outro length.

Reply:
This was a typo.

Reply:
Ah, that makes more sense now. Though the typo I actually meant was is still present `toDeckFadeDisatance` -> `toDeckFadeDistance`

Reply:
Oh, yes. Fixed as well.